doc/sphinx/user/methods/stokes-solver/solver.md

```diff
@@ -0,0 +1,222 @@
+# Stokes solver
+
+
+
+Problems in ASPECT require solving the Stokes
+problem below for velocity $u$ and pressure
+$p$
+
+```{math}
+:label: eq:strong-form
+-\nabla \cdot (2 \eta(x)\varepsilon(u))+\nabla p &= f \quad \text{ in } \Omega \\
+-\nabla \cdot u&=0 \quad\text{ in } \Omega
+```
+
+equipped with boundary conditions.
+
+
+Here, $x$ lies in our
+domain $\Omega$ and $\eta(x)$ is our
+viscosity. $\varepsilon(u)=\frac{1}{2}(\nabla u+
+\nabla u^T)$ is known as the strain rate tensor.
+
+In this document, we discuss the case
+of Dirichlet boundary conditions
+$u=g_D$ on $\Gamma_D$ and Neumann
+$n \cdot (pI-2\varepsilon(u))=g_N$ on
+$\Gamma_N$. However,
+ASPECT also supports
+Robin boundary conditions.
+
+
+To enable the
+computation of a numerical solution,
+and to loosen the regularity
+restrictions imposed on the velocity
+space by the strong formulation,
+we must first derive the weak form of
+the Stokes equations. This derivation
+can be found in step-22 of deal.II.
+
+To derive the weak form,
+let $u \in V_g:=\{\phi \in H^1(\Omega)^d: \phi_{\Gamma_D}=g_d\}$,
+$v \in V_0=\{\phi \in H^1(\Omega)^d: \phi_{\Gamma_D}=0\}$. We call $V_g$ the solution
+space and $V_0$ the test space.
+
+Multiplying the first equation in the strong form by $v \in V_0$ and integrating by parts:
+
+```{math}
+:label: eq:stokes-weak-form-1
+\begin{aligned}
+(-\nabla \cdot (2 \eta(x)\varepsilon(u)),v)+(\nabla p,v)&=
+(2\eta(x)\varepsilon(u),\nabla v)-(n \otimes v,2\eta(x)\varepsilon(u))_{d\Omega}\\
+&-(p,\nabla \cdot v)+(n \cdot v,p)_{d\Omega}\\
+&=(2\eta(x)\varepsilon(u),\varepsilon(v))-(p,\nabla \cdot v)+(v,n \cdot (pI-2\varepsilon(u))_{\Gamma_N})\\
+&=(f,v),
+\end{aligned}
+```
+
+where we use that $(n \otimes v,2\eta(x)\varepsilon(u))_{\Gamma_D}+(n\cdot v,p)_{\Gamma_D}=0$
+due to the definition of $V_0$, and the observation that
+$(2\eta(x)\varepsilon(u),\nabla v)=(2\eta(x)\varepsilon(u),\varepsilon(v))$.
+
+Of course, the $\nabla \cdot u =0$ we must also be
+incorporated into the
+weak formulation. Multiplying by $q \in Q=L^2$,
+we get $(q,\nabla \cdot u)=0$.
+
+Combining these two results,
+we have the weak form
+
+```{math}
+:label: eq:stokes-weak-form-2
+\begin{aligned}
+(\varepsilon(v),2\varepsilon(u))-(\nabla \cdot v,p)
+-(q,\nabla \cdot u)&=(v,f)-(v,g_N)_{\Gamma_N}
+\end{aligned}
+```
+where we apply the Neumann boundary
+condition to get the $(v,g_N)_{\Gamma_N}$ term
+from $(v,n \cdot(pI-2\varepsilon(u)_{\Gamma_N}))$.
+
+We proceed now to the
+discrete weak form. In order
+to guarantee existence and uniqueness
+of the discrete velocity and
+pressure solution, we
+search for $u_h \in Q_{k+1}^d$
+and $p_h \in Q_k$, with $k \geq 1$. The use
+$Q_{k+1}$ for velocity and  $Q_k$ for pressure for $k \geq 1$ satisfies
+the inf-sup condition
+and is typically known as the Taylor-Hood pair.
```

Comment:
leave a note that ASPECT also supports other discretizations (discontinuous pressure, q1q1 stabilized)